arelle/ModelObject.py

```diff
@@ -119,6 +119,7 @@ class ModelObject(etree.ElementBase):
     def _init(self) -> None:
         self.isChanged = False
         parent = self.getparent()
+        self.objectIndex = 0
```

Comment:
I'm not sure why this is needed. Is this initial value required for some reason? 0 is a valid index, so I'm a little concerned that this could cause a subtle bug in the future

Reply:
The code has been changed. 

Sometimes `objectIndex` is not defined.

 We have added an explicit check where it raised an exception.